src/components/transport_manager/src/usb/libusb/usb_connection.cc

```diff
@@ -202,28 +202,31 @@ bool UsbConnection::PostOutTransfer() {
 void UsbConnection::OnOutTransfer(libusb_transfer* transfer) {
   LOG4CXX_TRACE(logger_, "enter with  Libusb_transfer*: " << transfer);
   sync_primitives::AutoLock locker(out_messages_mutex_);
-  if (transfer->status == LIBUSB_TRANSFER_COMPLETED) {
-    bytes_sent_ += transfer->actual_length;
-    if (bytes_sent_ == current_out_message_->data_size()) {
-      LOG4CXX_DEBUG(
+  
+  if(current_out_message_.valid()) {
+    if (transfer->status == LIBUSB_TRANSFER_COMPLETED) {
```

Comment:
@BrandonHe It is better to  keep const on left side of comparator to avoid assign in case of mistake (`=` agains `==`)